kernel/src/drivers/panic.rs

```diff
@@ -0,0 +1,18 @@
+use core::panic::PanicInfo;
+use lock_api::RawMutex;
+use vexide_core::io::Write;
+
+use super::uart::UART1;
+
+#[panic_handler]
+fn panic_handler(info: &PanicInfo<'_>) -> ! {
+    critical_section::with(|_| {
+        unsafe {
+            // SAFETY: The UART device will not be used after the panic has been printed
+            // so we can consider the previous lock to be elapsed.
+            UART1.raw().unlock();
+        }
+        writeln!(UART1.try_lock().unwrap(), "Panic: {}", info).unwrap();
```

Comment:
So this seems fine, although I think a better write here could just be `"kernel {}"` which in practice reads as:
`kernel panicked at <...>:`

That way it's clear where the panic originates from at a quick glance.

versus `Panic: panicked at <...>:` which doesn't really provide much additional info about the nature of the panic compared to just logging `info`.